src/Shared/SqlProcessor.cs

```diff
@@ -228,177 +680,241 @@ private static bool SanitizeNumericLiteral(string sql, ref int index)
+
+        static SqlKeywordInfo()
         {
-            builder.Append(ch);
+            // Phase 1: Create all static instances.
+            // We will compare the SQL we are parsing in lowercase, so we store these in lowercase also.
+            AlterKeyword = new("alter", SqlKeyword.Alter, Unknown);
+            ClusteredKeyword = new("clustered", SqlKeyword.Clustered, [SqlKeyword.Unique]);
+            CreateKeyword = new("create", SqlKeyword.Create, Unknown);
+            DatabaseKeyword = new("database", SqlKeyword.Database, DdlKeywords);
+            DeleteKeyword = new("delete", SqlKeyword.Delete, Unknown);
+            DistinctKeyword = new("distinct", SqlKeyword.Distinct, [SqlKeyword.Select]);
+            DropKeyword = new("drop", SqlKeyword.Drop, Unknown);
+            ExistsKeyword = new("exists", SqlKeyword.Exists);
+            FromKeyword = new("from", SqlKeyword.From);
+            FunctionKeyword = new("function", SqlKeyword.Function, DdlKeywords);
+            IfKeyword = new("if", SqlKeyword.If);
+            IndexKeyword = new("index", SqlKeyword.Index, [.. DdlKeywords, SqlKeyword.Unique, SqlKeyword.Clustered, SqlKeyword.NonClustered]);
+            InsertKeyword = new("insert", SqlKeyword.Insert, Unknown);
+            IntoKeyword = new("into", SqlKeyword.Into);
+            JoinKeyword = new("join", SqlKeyword.Join);
+            NonClusteredKeyword = new("nonclustered", SqlKeyword.NonClustered, [SqlKeyword.Unique]);
+            NotKeyword = new("not", SqlKeyword.Not);
+            OnKeyword = new("on", SqlKeyword.On);
+            ProcedureKeyword = new("procedure", SqlKeyword.Procedure, DdlKeywords);
+            RoleKeyword = new("role", SqlKeyword.Role, DdlKeywords);
+            SchemaKeyword = new("schema", SqlKeyword.Schema, DdlKeywords);
+            SelectKeyword = new("select", SqlKeyword.Select, [SqlKeyword.Select, SqlKeyword.Unknown]);
+            SequenceKeyword = new("sequence", SqlKeyword.Sequence, DdlKeywords);
+            TableKeyword = new("table", SqlKeyword.Table, DdlKeywords);
+            TriggerKeyword = new("trigger", SqlKeyword.Trigger, DdlKeywords);
+            UnionKeyword = new("union", SqlKeyword.Union);
+            UniqueKeyword = new("unique", SqlKeyword.Unique, DdlKeywords);
+            UnknownKeyword = new(string.Empty, SqlKeyword.Unknown);
+            UpdateKeyword = new("update", SqlKeyword.Update, Unknown);
+            UserKeyword = new("user", SqlKeyword.User, DdlKeywords);
+            ViewKeyword = new("view", SqlKeyword.View, DdlKeywords);
+
+            // Phase 2: Build arrays that depend on instances
+            DdlSubKeywords = [
+                TableKeyword, IndexKeyword, ViewKeyword, ProcedureKeyword, TriggerKeyword,
+                DatabaseKeyword, SchemaKeyword, FunctionKeyword, UserKeyword, RoleKeyword, SequenceKeyword, UniqueKeyword,
+                ClusteredKeyword, NonClusteredKeyword
+            ];
```

Comment:
Do the order have significance? If not, can we sort them A-Z?

```suggestion
            DdlSubKeywords =
            [
                TableKeyword,
                IndexKeyword,
                ViewKeyword,
                ProcedureKeyword,
                TriggerKeyword,
                DatabaseKeyword,
                SchemaKeyword,
                FunctionKeyword,
                UserKeyword,
                RoleKeyword,
                SequenceKeyword,
                UniqueKeyword,
                ClusteredKeyword,
                NonClusteredKeyword,
            ];
```

Reply:
Yeah, order matters. I've tried to sort by the most likely although only based on my best guess. Arguably unique, clustered, nonclustered could probably move under procedure. Doing stuff with users, roles etc is likely rare is most application, whereas working with indexes may be more common. All are likely relatively rare scenarios TBH.